src/xal_be_fiemap.c

```diff
@@ -423,6 +423,8 @@ xal_be_fiemap_index(struct xal *xal)
 		return -EINVAL;
 	}
 
+	xal_pool_clear(&xal->inodes);
```

Comment:
Don't you need to also clear the extent-pool ``xal->extents``?

Also, there is a typo in the commit-message ``xak_index``